bridge/tracker/tracker.go

```diff
@@ -0,0 +1,390 @@
+		logger.Error("cannot load db", "err", err)
+
+		return nil, err
+	}
+
+	tracker.db = db
+
+	return tracker, nil
+}
+
+//	Start runs the tracker's listening mechanism:
+//	1. startEventTracking - goroutine keeping track of
+//	the last block queried, as well as collecting events
+//	of interest.
+//
+//	2. startSubscription - goroutine responsible for handling
+//	the subscription object. This object is provided by the
+//	client's subscribeNewHeads method that initiates the subscription.
+func (t *Tracker) Start() error {
+	//	create cancellable contexts
+	ctxSubscription, cancelSubscription := context.WithCancel(context.Background())
+	t.cancelSubscription = cancelSubscription
+
+	ctxEventTracking, cancelEventTracking := context.WithCancel(context.Background())
+	t.cancelEventTracking = cancelEventTracking
+
+	//	start event tracking process early
+	go t.startEventTracking(ctxEventTracking)
+
+	//	create newHeads subscription
+	subscription, err := t.client.subscribeNewHeads(t.headerCh)
+	if err != nil {
+		t.logger.Error("cannot subscribe to rootchain", "err", err)
+
+		return err
+	}
+
+	//	start receiving new header events
+	go t.startSubscription(ctxSubscription, subscription)
+
+	return nil
+}
+
+//	Stop stops the tracker's listening mechanism.
+func (t *Tracker) Stop() error {
+	//	stop subscription
+	t.cancelSubscription()
+
+	// 	stop processing headers
+	t.cancelEventTracking()
+
+	//	close rootchain client
+	if err := t.client.close(); err != nil {
+		t.logger.Error("cannot close rootchain client", "err", err)
+
+		return err
+	}
+
+	return nil
+}
+
+//	getEventChannel returns the tracker's event channel.
+func (t *Tracker) getEventChannel() <-chan []byte {
+	return t.eventCh
+}
+
+//	startSubscription handles the subscription object (provided by the rootchain client).
+func (t *Tracker) startSubscription(ctx context.Context, sub subscription) {
+	for {
+		select {
+		case err := <-sub.err():
+			t.logger.Error("subscription error", err)
+
+			t.cancelSubscription()
+			t.logger.Debug("cancelling subscription")
+
+			return
+		case <-ctx.Done():
+			if err := sub.unsubscribe(); err != nil {
+				t.logger.Error("cannot unsubscribe", "err", err)
+
+				return
+			}
+
+			t.logger.Debug("subscription stopped")
+
+			return
+		}
+	}
+}
+
+//	startEventTracking tracks each header (received by the subscription) for events.
+func (t *Tracker) startEventTracking(ctx context.Context) {
+	for {
+		select {
+		case header := <-t.headerCh:
+			t.trackHeader(header)
+		case <-ctx.Done():
+			t.logger.Debug("stopping header process")
+
+			return
+		}
+	}
+}
+
+//	loadDB creates a new database (or loads existing)
+//	for storing the last processed block's number by the tracker.
+func (t *Tracker) loadDB() (*leveldb.DB, error) {
+	//	get path
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	//	create or load db
+	db, err := leveldb.OpenFile(cwd+"/event_tracker/last_block_number", nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
`event_tracker` directory will be created under current directory. You might want to move this directory into data directory

Reply:
Good catch, missed this 💯

Reply:
The db folder should go where exactly ? As in, where should it reside in the project directory.